---
layout: post
title: "一些新的技巧"
date: 2019-2-27 19:00：00 +0300
categories:
- flask,Python
tags:
- flask,Python
---

# 1 关于re-raise的技巧

在捕捉异常的时候，如果不能直接抛出，想要保留异常的堆栈信息，是不能简单使用

```Python
try:
    1/0
except ZeroDivisionError as e:
    print(e)
```

这样做会丢失堆栈信息，让debug无从下手，
想要保存堆栈信息可以使用 traceback 库，优雅的做法是

```python
def dev():
    1/0

try:
    dev()
except ZeroDivisionError as e:
    msg = traceback.format_exc()
    print(msg)
```

结果：
```
Traceback (most recent call last):
  File "d:\Code\Python\normal_test\taaceback_test.py", line 7, in <module>
    dev()
  File "d:\Code\Python\normal_test\taaceback_test.py", line 4, in dev
    1/0
ZeroDivisionError: division by zero
```